### PR TITLE
utahfs: init at 1.0

### LINKED ISSUE #90338: Package Request: UtahFS
**Project description**
UtahFS is a state-of-the art encrypted storage solution, meant to be similar to Dropbox. It has a FUSE binding that creates a synthetic drive on the user's computer that they can interact with like an external hard-drive. Files stored in the drive are uploaded to a cloud storage provider, which means the drive will never run out of space and minimizes the likelihood of any files being lost. However the files are encrypted such that the cloud storage provider knows almost nothing about what's being stored.

Completely written in *Go*.

**Metadata**

* homepage URL: https://github.com/cloudflare/utahfs
* source URL: https://github.com/cloudflare/utahfs
* license: bsd
* platforms: unix

### PR DESCRIPTION
**Pull request by snglth**
###### Motivation for this change

Fixes: #90338

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/utahfs/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  goPackagePath = "github.com/cloudflare/utahfs";
+  goDeps = ./deps.nix;
+
+  meta = with lib; {
+    homepage = "https://github.com/cloudflare/utahfs";
+    description =
+      "Encrypted storage system that provides a user-friendly FUSE drive backed by cloud storage";
+    license = licenses.bsd3;
+    maintainers = [ maintainers.snglth ];
```
Comment (mdlayher) [inline pkgs/applications/networking/utahfs/default.nix:21]: Nit: I see `with maintainers; [ snglth ];` used more frequently, but no big deal.

**Review by mdlayher — APPROVED** [review]
LGTM

**Comment by mdlayher** [comment]
Builds just fine:

```
[matt@servnerr-3:~/src/nixpkgs]$ nixpkgs-review pr 91060
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/91060/head:refs/nixpkgs-review/1
remote: Enumerating objects: 17, done.
remote: Counting objects: 100% (17/17), done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 21 (delta 9), reused 8 (delta 8), pack-reused 4
Unpacking objects: 100% (21/21), 13.49 KiB | 767.00 KiB/s, done.
From https://github.com/NixOS/nixpkgs
 * [new branch]              master               -> refs/nixpkgs-review/0
 * [new ref]                 refs/pull/91060/head -> refs/nixpkgs-review/1
$ git worktree add /home/matt/.cache/nixpkgs-review/pr-91060/nixpkgs 4e69b0a89934d3516b72ca2175fabe85f6f540d7
Preparing worktree (detached HEAD 4e69b0a8993)
Updating files: 100% (21934/21934), done.
HEAD is now at 4e69b0a8993 Merge pull request #89744 from vojta001/traefik-indent-config
$ git merge --no-commit 816524bbc5bdc1ba427f1e5115d532d83fa9223d
Auto-merging pkgs/top-level/all-packages.nix
Automatic merge went well; stopped before committing as requested
$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /home/matt/.cache/nixpkgs-review/pr-91060/build.nix
warning: ignoring the user-specified setting 'sandbox', because it is a restricted setting and you are not a trusted user
[22 built, 19 copied (96.3 MiB), 12.3 MiB DL]
https://github.com/NixOS/nixpkgs/pull/91060
1 package built:
utahfs

[0.0 MiB DL]
$ nix-shell /home/matt/.cache/nixpkgs-review/pr-91060/shell.nix
these paths will be fetched (1.51 MiB download, 8.27 MiB unpacked):
  /nix/store/3fd40kcri44gl9nkzn75krvh2rpgbp3n-bash-interactive-4.4-p23-dev
  /nix/store/amh412bj67yaaaiqyh58h22jp7zh8x60-readline-7.0p5
  /nix/store/gr4nh9lch6dn9axs82jmv0bbs7li4zxa-bash-interactive-4.4-p23-doc
  /nix/store/hgplspyw16amnbhlbi7qa2938w8i0rz6-bash-interactive-4.4-p23
  /nix/store/hrni2kgw4m661b65h3lv28hvbp79az24-bash-interactive-4.4-p23-man
  /nix/store/wd9xmxvsv5gfd71c1c7mxja92wlprbhl-bash-interactive-4.4-p23-info
copying path '/nix/store/gr4nh9lch6dn9axs82jmv0bbs7li4zxa-bash-interactive-4.4-p23-doc' from 'https://cache.nixos.org'...
copying path '/nix/store/wd9xmxvsv5gfd71c1c7mxja92wlprbhl-bash-interactive-4.4-p23-info' from 'https://cache.nixos.org'...
copying path '/nix/store/hrni2kgw4m661b65h3lv28hvbp79az24-bash-interactive-4.4-p23-man' from 'https://cache.nixos.org'...
copying path '/nix/store/amh412bj67yaaaiqyh58h22jp7zh8x60-readline-7.0p5' from 'https://cache.nixos.org'...
copying path '/nix/store/hgplspyw16amnbhlbi7qa2938w8i0rz6-bash-interactive-4.4-p23' from 'https://cache.nixos.org'...
copying path '/nix/store/3fd40kcri44gl9nkzn75krvh2rpgbp3n-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...

[nix-shell:~/.cache/nixpkgs-review/pr-91060]$ ./results/utahfs/bin/utahfs-server -h
Usage of ./results/utahfs/bin/utahfs-server:
  -cfg string
        Location of the server's config file. (default "./utahfs.yaml")
  -metrics-addr string
        Address to serve metrics on. (default "localhost:3003")
  -server-addr string
        Address to expose server on. (default "0.0.0.0:3002")

```

File: pkgs/applications/networking/utahfs/default.nix (hunk 1 above)
Comment (snglth) [reply pkgs/applications/networking/utahfs/default.nix:21]: It's really pointless if there is only one maintainer or whatever to be put into expression's lexical scope, IMHO.

**Comment by devhell** [comment]
Amazing, thank you!

File: pkgs/applications/networking/utahfs/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, buildGoPackage, lib, fetchFromGitHub }:
```
Comment (veprbl) [inline pkgs/applications/networking/utahfs/default.nix:1]:
```suggestion
{ lib, buildGoPackage, fetchFromGitHub }:
```
`stdenv` is not used

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -26839,4 +26839,6 @@ in
   vpsfree-client = callPackage ../tools/virtualization/vpsfree-client {};
 
   gpio-utils = callPackage ../os-specific/linux/kernel/gpio-utils.nix { };
+
+  utahfs = callPackage ../applications/networking/utahfs { };
```
Comment (veprbl) [inline pkgs/top-level/all-packages.nix:26843]: Please, don't add to the end of the file. Instead, I suggest you locate an appropriate section (e.g. using `grep '###' pkgs/top-level/all-packages.nix`, probably `### TOOLS` or `### APPLICATIONS`) and insert your attribute according to the alphabetic order. This should also make merge conflicts unlikely.

**Review by veprbl — COMMENTED** [review]
Looks fine, but need to resolve the merge conflict.

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (snglth) [reply pkgs/top-level/all-packages.nix]: Done. Thak you for the review!

**Review by veprbl — APPROVED** [review]

**Pull request merged**